builders/interoperability/xcm/remote-evm-calls.md

```diff
@@ -7,34 +7,33 @@ description: 如何通过XCM从任何已建立XCM通道的波卡平行链远程
+为解决此问题，引入了[Ethereum XCM Pallet](https://github.com/PureStake/moonbeam/tree/master/pallets/ethereum-xcm){target=_blank}。它充当XCM [Transact](https://github.com/paritytech/xcm-format#transact){target=_blank}指令和[Ethereum Pallet](https://github.com/paritytech/frontier/tree/master/frame/ethereum){target=_blank}之间的中间件，因为在通过XCM远程执行EVM调用时需要特别注意。Pallet将执行必要的检查并验证交易。接着，Pallet会调用Ethereum Pallet将交易派遣给EVM。根据访问EVM的方式，常规和远程EVM的调用之间存在一些差异。
 
 下列流程图描绘了通过XCM进行常规和远程EVM调用的路径：
 
 ![Happy parth for regular and remote EVM calls through XCM](/images/builders/interoperability/xcm/remote-evm-calls/xcmevm-1.png)
 
-本教程将介绍常规和远程EVM调用之间的差异。此外将展示如何通过[Ethereum-XCM pallet](https://github.com/PureStake/moonbeam/tree/master/pallets/ethereum-xcm){target=_blank}中函数执行远程EVM调用。
+本教程将介绍常规和远程EVM调用之间的差异。此外将展示如何通过[Ethereum XCM Pallet](https://github.com/PureStake/moonbeam/tree/master/pallets/ethereum-xcm){target=_blank}中函数执行远程EVM调用。
 
 !!! 注意事项
-    远程EVM调用通过[XCM-transactor pallet](/builders/interoperability/xcm/xcm-transactor/){target=_blank}完成。因此，建议您在尝试通过XCM执行远程EVM调用之前熟悉XCM-transactor概念。
+    远程EVM调用通过[XCM Transactor Pallet](/builders/interoperability/xcm/xcm-transactor/){target=_blank}完成。因此，建议您在尝试通过XCM执行远程EVM调用之前熟悉XCM-transactor概念。
 
 **请注意，通过XCM对Moonbeam执行EVM的远程调用仍在积极开发中**。此外，**开发人员必须了解，发送不正确的XCM消息将导致资金损失。**因此，在迁移到生产环境之前须在测试网上测试XCM功能。
 
 ## 相关XCM定义 {: #general-xcm-definitions }
 
 --8<-- 'text/xcm/general-xcm-definitions2.md'
 
- - **衍生账户** — 从另一个帐户衍生的帐户。衍生账户是无密钥的，也就是私钥未知。因此，与XCM特定用例相关的衍生账户只能通过XCM相关的extrinsics访问。对于远程的EVM调用，主要类型如下：
-     - **Multilocation衍生账户** — 这会生成一个无密钥帐户，该帐户从由 [`DescendOrigin`](https://github.com/paritytech/xcm-format#descendorigin){target=_blank} XCM指令和提供的Multilocation设置的新来源所衍生。对于基于Moonbeam的网络，[衍生函数](https://github.com/PureStake/moonbeam/blob/master/primitives/xcm/src/location_conversion.rs#L31-L37){target=_blank}计算Multilocation的`blake2`哈希，包括原始平行链ID，并截取正确长度的哈希（以太坊格式的帐户为20个字节）。XCM调用[原转换](https://github.com/paritytech/polkadot/blob/master/xcm/xcm-executor/src/lib.rs#L343){target=_blank}在`Transact`指令执行时发生。因此，每条平行链都可以使用自己想要的程序转换来源，因此发起交易的用户可能在每条平行链上拥有不同的衍生账户。该衍生账户用于支付交易费用，并被设置为调用的调度者
- - **交换信息** — 与XCM-transactor extrinsic的XCM远程执行部分的额外权重和费用信息有关。这部分为必要的，因主权账户将支付XCM交易费用。因此，XCM-transactor就计算费用数值，并向XCM-transactor extrinsic的发送方收取相应[XC-20 Token](/builders/interoperability/xcm/xc20/overview/){target=_blank}的估计数量以偿还主权账户
+- **Multilocation衍生账户** — 这会生成一个无密钥帐户，该帐户从由 [`DescendOrigin`](https://github.com/paritytech/xcm-format#descendorigin){target=_blank} XCM指令和提供的Multilocation设置的新来源所衍生。对于基于Moonbeam的网络，[衍生函数](https://github.com/PureStake/moonbeam/blob/master/primitives/xcm/src/location_conversion.rs#L31-L37){target=_blank}计算Multilocation的`blake2`哈希，包括原始平行链ID，并截取正确长度的哈希（以太坊格式的帐户为20个字节）。XCM调用[原转换](https://github.com/paritytech/polkadot/blob/master/xcm/xcm-executor/src/lib.rs#L343){target=_blank}在`Transact`指令执行时发生。因此，每条平行链都可以使用自己想要的程序转换来源，因此发起交易的用户可能在每条平行链上拥有不同的衍生账户。该衍生账户用于支付交易费用，并被设置为调用的调度者
```

Comment:
```suggestion
- **Multilocation衍生账户** — 这会生成一个无密钥帐户，该帐户从由 [`DescendOrigin`](https://github.com/paritytech/xcm-format#descendorigin){target=_blank} XCM指令和提供的Multilocation设置的新来源所衍生。对于基于Moonbeam的网络，[衍生函数](https://github.com/moonbeam-foundation/moonbeam/blob/v0.31.1/primitives/xcm/src/location_conversion.rs#L31-L37){target=_blank}计算Multilocation的`blake2`哈希，包括原始平行链ID，并截取正确长度的哈希（以太坊格式的帐户为20个字节）。XCM调用[原转换](https://github.com/paritytech/polkadot/blob/master/xcm/xcm-executor/src/lib.rs#L343){target=_blank}在`Transact`指令执行时发生。因此，每条平行链都可以使用自己想要的程序转换来源，因此发起交易的用户可能在每条平行链上拥有不同的衍生账户。该衍生账户用于支付交易费用，并被设置为调用的调度者
```

Reply:
Good catch! 

I'll accept this suggestion, but we ultimately need to change the text here because now we don't have a custom solution for calculating multilocation derivative accounts.